test/integration/targets/iam_user/tasks/main.yml

```diff
@@ -0,0 +1,119 @@
+---
+- name: set up aws connection info
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+
+  - name: ensure ansible user exists
+    iam_user:
+      name: '{{ test_user }}'
+      state: present
+
+  - name: get info on IAM user(s)
+    iam_user_info:
+    register: iam_user_info
+
+  - name: assert correct number of users on path are returned
+    assert:
+      that:
+        - iam_user_info.iam_users | length != 0
+
+  - name: get info on IAM user(s) with name
+    iam_user_info:
+      name: '{{ test_user }}'
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info.iam_users | length == 1
+        - 'iam_user_info.iam_users[0].arn.startswith("arn:aws:iam")'
+        - 'iam_user_info.iam_users[0].arn.endswith("user/" + test_user )'
+        - iam_user_info.iam_users[0].createDate is not none
+        - iam_user_info.iam_users[0].path == '{{ test_path }}'
+        - iam_user_info.iam_users[0].user_id is not none
+        - iam_user_info.iam_users[0].user_name == '{{ test_user }}'
+
+  - name: get info on IAM user(s) on path
+    iam_user_info:
+      path: '{{ test_path }}'
+      name: '{{ test_user }}'
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info.iam_users | length == 1
+        - 'iam_user_info.iam_users[0].arn.startswith("arn:aws:iam")'
+        - 'iam_user_info.iam_users[0].arn.endswith("user/" + test_user )'
+        - iam_user_info.iam_users[0].createDate is not none
+        - iam_user_info.iam_users[0].path == '{{ test_path }}'
+        - iam_user_info.iam_users[0].user_id is not none
+        - iam_user_info.iam_users[0].user_name == '{{ test_user }}'
+
+  - name: ensure group exists
+    iam_group:
+      name: '{{ test_group }}'
+      users:
+        - '{{ test_user }}'
+      state: present
+    register: iam_group
+
+  - name: get info on IAM user(s) in group
+    iam_user_info:
+      group: '{{ test_group }}'
+      name: '{{ test_user }}'
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info.iam_users | length == 1
+        - 'iam_user_info.iam_users[0].arn.startswith("arn:aws:iam")'
+        - 'iam_user_info.iam_users[0].arn.endswith("user/" + test_user )'
+        - iam_user_info.iam_users[0].createDate is not none
+        - iam_user_info.iam_users[0].path == '{{ test_path }}'
+        - iam_user_info.iam_users[0].user_id is not none
+        - iam_user_info.iam_users[0].user_name == '{{ test_user }}'
+
+  - name: attempt using both a path and group
+    iam_user_info:
+      path: '{{ test_path }}'
+      group: '{{ test_group }}'
+      name: '{{ test_user }}'
+    ignore_errors: yes
+    register: iam_user_info_path_group
+
+  - name: assert improper usage of parameters fails gracefully
```

Comment:
I tend to drop the parameter tests right at the top, they're quick and failure means you don't need to create anything.

Reply:
https://github.com/ansible/ansible/pull/23382/commits/510e22e39c839f96c157306b19f0a2563dbbe2ab moves this.